shell/browser/electron_browser_main_parts.cc

```diff
@@ -217,6 +319,21 @@ int ElectronBrowserMainParts::PreEarlyInitialization() {
   HandleSIGCHLD();
 #endif
 #if BUILDFLAG(IS_LINUX)
+  auto* const command_line = base::CommandLine::ForCurrentProcess();
+  if (!command_line->HasSwitch(switches::kOzonePlatform)) {
+    const auto ozone_platform_hint =
+        command_line->GetSwitchValueASCII(switches::kOzonePlatformHint);
+    if (!ozone_platform_hint.empty()) {
+      command_line->AppendSwitchASCII(
+          switches::kOzonePlatform, MaybeFixPlatformName(ozone_platform_hint));
+    }
+  }
+
+  auto env = base::Environment::Create();
+  std::string desktop_startup_id;
+  if (env->GetVar("DESKTOP_STARTUP_ID", &desktop_startup_id))
+    command_line->AppendSwitchASCII("desktop-startup-id", desktop_startup_id);
+
```

Comment:
I like the idea of keeping this file minimal. We can have a new file `electron_browser_main_parts_linux.cc` and move this logic to a new helper method `void DetectOzonePlatform()` and also move the other static helper functions `MaybeFixPlatformName`, `HasWaylandDisplay` into that file.

Reply:
These suggested changes have now been implemented.